tests/test_opam.ml

```diff
@@ -0,0 +1,27 @@
+let test_version =
+  let make_test ~input ~expected =
+    let name = "version " ^ input in
+    let test_fun () =
+      let actual =
+        match Dune_release.Opam.version input with
+        | Ok `v1_2_2 -> "v1_2_2"
```

Comment:
How about adding the right testables here so that we don't have to convert to strings? That seems like a cleaner use of `Alcotest`.

You can even extract a `version` type in `Opam` since there is no reason for this one to be a polymorphic variant.

Reply:
good idea, done